flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/partitioner/RebalancePartitioner.java

```diff
@@ -36,9 +38,13 @@
 	@Override
 	public int[] selectChannels(SerializationDelegate<StreamRecord<T>> record,
 			int numberOfOutputChannels) {
-		int newChannel = ++this.returnArray[0];
-		if (newChannel >= numberOfOutputChannels) {
-			this.returnArray[0] = 0;
+		if (this.returnArray[0] < 0) {
+			this.returnArray[0] = new Random().nextInt(numberOfOutputChannels);
```

Comment:
May be use  `ThreadLocalRandom.current().nextInt` would be better than `new Random().nextInt`. Even if you use `Random` class, there is no need to create it every time, when calling this method.

Reply:
Thanks for your suggestion, and I am going to commit an update.